API/Startup.cs

```diff
@@ -235,19 +235,25 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
                         string identityId = context.User.GetIdentityId(context);
                         if(await userService.GetUserByIdentityIdAsync(identityId).ConfigureAwait(false) == null)
                         {
+                            IRoleService roleService = context.RequestServices.GetService<IRoleService>();
+                            Role registeredUserRole = (await roleService.GetAll()).FirstOrDefault(i => i.Name == nameof(Defaults.Roles.RegisteredUser));
+
                             User newUser = context.GetUserInformation(Config);
                             if(newUser == null)
                             {
+
                                 // Then it probably belongs swagger so we set the username as developer.
                                 newUser = new User()
```

Comment:
You create the user if it doesn't exist? and if it doesn't exist you just assume it is swagger? Seems odd to me

Reply:
Swagger does not support the full openid spec, mainly the id_token.
so when a user as a token instead of an id token it will set its name and email to developer.
the frontend can request an id token and then we can request the users information from the identity server.